src/map_calc.cpp

```diff
@@ -456,6 +456,26 @@ NumericMatrix pop_tally(IntegerMatrix districts, vec pop, int n_distr) {
     return tally;
 }
 
+/*
+ * Create the projective distribution of a variable `x`
+ */
+// [[Rcpp::export]]
+NumericMatrix proj_distr_m(IntegerMatrix districts, const arma::vec x,
+                           IntegerVector draw_idx, int n_distr) {
+    int n = draw_idx.size();
+    int V = districts.nrow();
+
+    NumericMatrix out(V, n);
+    for (int i = 0; i < n; i++) {
+        for (int j = 0; j < V; j++) {
+            int idx = draw_idx[i] - 1;
```

Comment:
Would it be better to define `int idx;` at the top?

Reply:
why? either way the compiler will translate to the most efficient LLVM code